contracts/settling_game/modules/relics/Relics.cairo

```diff
@@ -97,31 +112,145 @@ func set_relic_holder{pedersen_ptr: HashBuiltin*, syscall_ptr: felt*, range_chec
         tempvar pedersen_ptr = pedersen_ptr;
     }
 
-    // Capture back if held by loser
-    // Check if loser has winners token
-    let (winners_relic_owner) = get_current_relic_holder(winner_token_id);
+    let (realms_address) = IModuleController.get_external_contract_address(
+        controller, ExternalContractIds.Realms
+    );
+
+    let (winner_realm_data: RealmData) = IRealms.fetch_realm_data(realms_address, winner_token_id);
+
+    let (loser_relics_len) = owned_relics_count.read(loser_token_id);
```

Comment:
This value, the number of relics owned by the loser, is never decremented. Implies that the `owned_relics` array is never purged of 0 values. It will only grow, eventually looking like e.g. `[0, 0, 0, ....., 782]` if I'm not mistaken.

Reply:
Yes this is the current case, something I mentioned at the start of the PR. I can't see a way around this currently, as we need to keep track of all relics owned where a relic will be removed at a random index in an array.

E.g removing index `1` from `[1,2,3]` leaves `[1,0,3]`. We cannot decrement the count as we would then only be looping on `[1,0]` where `3` still exists in the array.

Any suggestions of how we can get around this 🤔

Reply:
Now I think of it perhaps just creating a new array. Looping over the original and only pushing the value when it is bigger than 0. Then storing this over the original array storage variable.

Reply:
This is a problem for an infinite game. We can't merge it as is.

Maybe using some different data model would help? Or doing some bit packing? I don't have a clear solution in mind. I would also be in favor of altering the whole mechanic TBH. We have to work within the constraints of the system.

Reply:
We could move the data off-chain - so the victor needs to enter the Relics they want to capture back